lib/AST/Module.cpp

```diff
@@ -165,19 +165,23 @@ class swift::SourceLookupCache {
   using OperatorMap = llvm::DenseMap<Identifier, TinyPtrVector<T *>>;
   OperatorMap<OperatorDecl> Operators;
   OperatorMap<PrecedenceGroupDecl> PrecedenceGroups;
+  SmallVector<MacroExpansionDecl *, 4> DelayedMacroExpansions;
```

Comment:
I agree that name-driven lazy expansion is a better approach, and was going to do it with a follow-up patch after you implemented it global peers to avoid conflicts

Reply:
Implemented name-driven expansion!